<?xml version="1.0"?>
<launch>
    <arg name="play_bag" default="true"/>
    <arg name="bag_file_name" default="$(find analysis_mp_chaser)/world/los_keeper_simulation_2d.bag"/>
    <arg name="rviz_display" default="true"/>
    <arg name="run_planner" default="false"/>

    <group if="$(arg play_bag)">
        <node name="ros_bag_play" pkg="rosbag" type="play" args="--clock -r 1.0 -s 0.0 $(arg bag_file_name)"/>
    </group>
    <group if="$(arg rviz_display)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find analysis_mp_chaser)/rviz_config/los_keeper_2d.rviz"/>
    </group>

    <node pkg="analysis_mp_chaser" type="analyzer_los_keeper" name="analyzer" output="screen">
        <param name="is_2d" value="true"/>
        <param name="write_total_trajectory" value="false"/>
        <param name="calculate_min_distance" value="false"/>
        <param name="min_distance_file_name" value="$(find analysis_mp_chaser)/log/min_distance_2d_sim.txt"/>
        <param name="calculate_visibility_score" value="false"/>
        <param name="visibility_score_file_name" value="$(find analysis_mp_chaser)/log/visibility_score_2d_sim.txt"/>
        <param name="total_trajectory_filename_write" value="$(find analysis_mp_chaser)/log/actor_trajectory_simulator_2d.txt"/>
        <param name="total_trajectory_filename_read" value="$(find analysis_mp_chaser)/data/actor_trajectory_simulator_2d.txt"/>
    </node>
    <group if="$(arg run_planner)">
        <node pkg="los_keeper" type="los_keeper" name="los_keeper" output="screen">
            <rosparam command="load" file="$(find los_keeper)/param/plan_2d_sc1_multi_thread.yaml"/>
            <rosparam command="load" file="$(find los_keeper)/param/visualization.yaml"/>
        </node>
    </group>

</launch>